sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -315,6 +320,10 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
       case join @ ExtractEquiJoinKeys(joinType, leftKeys, rightKeys, _, _, left, right, hint) =>
         var newLeft = left
         var newRight = right
+        // Whether it is a shuffle join or not should be based on the actual left and
+        // right table. For some join like left outer join, it will be a shuffle join
+        // even if left side table size is smaller than broadcast threshold.
+        val isShuffleJoin = isProbablyShuffleJoin(left, right, hint, joinType)
```

Comment:
I think we can pass joinType into `extractBeneficialFilterCreatePlan`, `isProbablyShuffleJoin` and `probablyHasShuffle`. Then we can avoid breaking the code structure.

Reply:
> I think we can pass joinType into `extractBeneficialFilterCreatePlan`, `isProbablyShuffleJoin` and `probablyHasShuffle`. Then we can avoid breaking the code structure.
@beliefer 
The decision if a join is shuffle or nor depends on the join type and also the size of table on left and right side of the join. In method extractBeneficialFilterCreatePlan, the joined tables are interchanged to check if any of the table can be used to build the bloom filter. So it is difficult to judge the left and right table in method extractBeneficialFilterCreatePlan. That's why the decision if the join is shuffle or not is done outside. Please let me know if it makes sense.

Reply:
I got it.